File: app/src/main/java/com/example/alarm/model/AlarmRepository.java
package com.example.alarm.model;

import android.content.Context;
import java.util.List;

public interface AlarmRepository {

    List<Alarm> getAlarms() throws Exception;

    boolean addAlarm(Alarm alarm) throws Exception;

    boolean updateAlarm(Alarm alarm) throws Exception;

    void updateEnabled(long id, Boolean enabled) throws Exception;

    void deleteAlarms(List<Alarm> list, Context context) throws Exception;
}

File: app/src/main/java/com/example/alarm/model/AlarmRepository.kt
package com.example.alarm.model

import android.content.Context
import kotlinx.coroutines.flow.Flow

interface AlarmRepository {

    suspend fun getAlarms(): MutableList<Alarm>

    suspend fun addAlarm(alarm: Alarm) : Boolean

    suspend fun updateAlarm(alarm: Alarm) : Boolean

    suspend fun updateEnabled(id: Long, enabled: Int)

    fun getAlarmById(id: Long): Flow<Alarm?>

    suspend fun deleteAlarms(list: List<Alarm>, context: Context?)
}